cmake: fix regex for west version check in host-tools.cmake

thst-nordic opened the pull request:
'west --version' output has the path to the west manifest directory along with the version.
If there is a semantic version string in this path name (like when CI tests a tag), it raises
a false version check failure.

New regex doesn't check version string inside path. 

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

SebastianBoe commented:
Hi Thomas, this "[nrf toup] " should not be used on commits that go to this repository.

SebastianBoe commented:
Perhaps eventually something like

west --version --machine-readable
west --version-without-path
west --install-dir

or similar would be appropriate.

thst-nordic committed:
cmake: fix regex for west version check in host-tools.cmake

'west --version' output has the path to the west manifest directory.
If there is a semantic version string in this path name, it will
raise a false version check failure.

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

thst-nordic commented:
- I removed nrf sauce.  
- I'll talk to Marti about --machine-readable.

mbolivar commented:
```
west --version --machine-readable
west --version-without-path
```

I think the current output is relatively easy to parse; the problem was my fault for using a greedy `.*` before searching for the `vX.Y.Z` when adding this in the first place.

```
west --install-dir
```

This is a separate question, no?

SebastianBoe commented:
> west --install-dir
> This is a separate question, no?

I imagined one would want west to just spit out the install directory of the components, and omit the versions for easier parsing. But if you deem the parsing to be acceptable then this is not necessary.

SebastianBoe commented:
@nashif : This is ready to be merged.

mbolivar commented:
> I imagined one would want west to just spit out the install directory of the components, and omit the versions for easier parsing. But if you deem the parsing to be acceptable then this is not necessary.

Fair. Though `pip3 show -f west | grep ^Location:` (or its CMake equivalent) serves this purpose already. Stripping the leading `Location: ` seems easy enough.
